src/prettytable/prettytable.py

```diff
@@ -33,6 +33,7 @@
 
 from __future__ import annotations
 
+import functools
```

Comment:
Let's use this form:
```suggestion
from functools import lru_cache
```